Sources/MongoSwift/MongoError.swift

```diff
@@ -2,277 +2,287 @@ import CLibMongoC
 import Foundation
 
 /// An empty protocol for encapsulating all errors that this package can throw.
-public protocol MongoError: LocalizedError {}
-
-// TODO: update this link and the one below (SWIFT-319)
-/// A MongoDB server error code.
-/// - SeeAlso: https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.yml
-public typealias ServerErrorCode = Int
+public protocol MongoErrorProtocol: LocalizedError {}
 
 /// Protocol conformed to by errors that may contain error labels.
-public protocol LabeledError: MongoError {
+public protocol MongoLabeledError: MongoErrorProtocol {
```

Comment:
There are limitations to this enum-as-namespace approach, namely protocols can't be defined that way. To avoid collisions, I added the `Mongo` prefix to all these protocols.